File: workflow-examples/transaction-monitoring/backend-typescript/src/services/compilot.ts
import { Transaction } from '../types/transaction';

export class ComPilotService {
    static async submitTransaction(transaction: Transaction) {
        const response = await fetch(`${process.env.COMPILOT_API_URL}/transactions`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${process.env.COMPILOT_API_KEY}`
            },
            body: JSON.stringify(transaction)
        });

        const data = await response.json();

        if (!response.ok) {
            throw new Error(`ComPilot API Error: ${JSON.stringify(data, null, 2)}`);
        }

        return data;
    }
} 

File: workflow-examples/transaction-monitoring/backend-python/services/compilot.py
import os
import requests
from dotenv import load_dotenv

load_dotenv()

class ComPilotService:
    @staticmethod
    def submit_transaction(transaction):
        try:
            response = requests.post(
                f"{os.getenv('COMPILOT_API_URL')}/transactions",
                json=transaction,
                headers={
                    'Content-Type': 'application/json',
                    'Authorization': f"Bearer {os.getenv('COMPILOT_API_KEY')}"
                }
            )
            
            response.raise_for_status()
            return response.json()
            
        except requests.exceptions.RequestException as e:
            if e.response is not None:
                error_data = e.response.json()
                raise Exception(f"ComPilot API Error: {error_data}")
            raise Exception(f"ComPilot API Error: {str(e)}") 